### ISSUE TITLE
States Stop Updating after Some Time

### ISSUE THREAD

**Issue by ztrglider** [custom-components/remote_homeassistant#33]
All entities synced perfectly when first set up the config and when I restart the server. However, after some time (sometimes an hour, sometimes four), the states stop updating on the master. The values are accurate on my remote instance.  I first noticed the issue with the climate domain, but from the history, I think all the remote entities stopped syncing at the same time. 

What information would help narrow in on this issue?

**Comment by ztrglider** [comment]
More info: I found this error in the logs...

```
Wed Jan 08 2020 10:29:57 GMT-0500 (Eastern Standard Time)
Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/remote_homeassistant/__init__.py", line 272, in _recv
    callback(message)
  File "/config/custom_components/remote_homeassistant/__init__.py", line 393, in fire_event
    self._entities.remove(entity_id)
KeyError: 'persistent_notification.config_entry_discovery'

```

**Comment by jacekpaszkowski** [comment]
@ztrglider i noticed same issue.  I've also found simillar erro in logs and i think that after this error all goes down.

```
2020-01-15 08:54:31 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/remote_homeassistant/__init__.py", line 272, in _recv
    callback(message)
  File "/config/custom_components/remote_homeassistant/__init__.py", line 393, in fire_event
    self._entities.remove(entity_id)
KeyError: 'persistent_notification.slave1_http_login'
```

There is exclude in component configuration:
```
      exclude:
        domains:
          - persistent_notification
```

**Comment by jacekpaszkowski** [comment]
Yes, i can confirm that component crashes just after this kind of error, just happend today

```
2020-01-21 11:24:12 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/remote_homeassistant/__init__.py", line 272, in _recv
    callback(message)
  File "/config/custom_components/remote_homeassistant/__init__.py", line 393, in fire_event
    self._entities.remove(entity_id)
KeyError: 'persistent_notification.slave1_google_fit_notification'
```

Any chances to fix that?

**Comment by lukas-hetzenecker** [comment]
Hm, the `with suppress(ValueError, AttributeError, KeyError)` on the line before should actually suppress this error. 
Could somebody that can reproduce this issue maybe try to replace the `with suppress` contextmanager, with a `try .. except: pass` block? I'm a bit confused here, how this exception can appear..

**Comment by jacekpaszkowski** [comment]
@lukas-hetzenecker I use latest relese (version 2.1) as i've checked in the line 393 there is
```with suppress(ValueError, AttributeError)``` no ```suppress(ValueError, AttributeError, KeyError)```.
I'll try switching to master version.

**Comment by jacekpaszkowski** [comment]
@lukas-hetzenecker i can confirm that master branch version works ok. No KeyError exceptions.

**Comment by rchamp26** [comment]
hi there. i've been testing this for some months now, but it looks like i am running into the same issue. i've tried with both the 2.1 branch and the master. 

works fine for about an hour or 2 after HA restarts, then stops providing updates. i've seen a couple errors from time to time about the websocket closing. 

i've enabled debug logs for this to see if i can capture more data to provide. currently i'm running HA version 0.112.3

i've also excluded persistent notifications as mentioned higher in the thread as i don't need that information.

**Comment by lukas-hetzenecker** [comment]
Hi people,
sorry I can't bring any better news, but I still haven't encountered this issue in my setup. 
And unfortunately currently I also don't have much time for working on this component (as you can probably guess from my activity), so still haven't figured out how to reproduce this issue.
Logs from Configuration -> Logs would probably be helpful though. Are there any exceptions logged from the remote_homeassistant component?

**Comment by rchamp26** [comment]
understood @lukas-hetzenecker. I had to adjust my logger to reduce the amount of logs to filter through. will post some error logs shortly. 

hopefully i can pull some meaningful logs that can point to where the problem lies

**Comment by rchamp26** [comment]
> understood @lukas-hetzenecker. I had to adjust my logger to reduce the amount of logs to filter through. will post some error logs shortly.
> 
> hopefully i can pull some meaningful logs that can point to where the problem lies

still waiting on errors. one change i made was to reduce what was coming from the remote instance. i didn't have many sensors on the remote instance, but did have a bunch of unused glances sensors, so i included only the specific sensors in question. since that change and restart on the main instance, i haven't seen the halt. I'll let it run today and report back tomorrow and see how things go ::fingers-crossed::

**Comment by rchamp26** [comment]
Well it's happening again, but I can't seem to get any logs. When set to debug, the info is way too overwhelming getting pretty much every update with thousands of lines to filter through. When I switch to error logging only, I don't seem to get any errors. My last test will be to spin up an empty Hass instance to connect these remote instances to try and see if it's something with the most recent hass or other integration that is causing the web sockets to close. If it doesn't stop updating on that instance, then it's something else withing my core Hass instance causing the web sockets to close and not recover.

**Comment by rchamp26** [comment]
the only clear error messages i can find is this 

```
2020-07-25 11:03:48 ERROR (MainThread) [custom_components.remote_homeassistant] websocket connection is closing
```

**Comment by postlund** [comment]
Anyone using version 3.0.0 that can confirm if this is still an issue?

**Comment by jacekpaszkowski** [comment]
i haven't noticed this king of problem for long time

**Comment by postlund** [comment]
@jacekpaszkowski Thanks for reporting! 👍

@rchamp26 Can you try to replicate with latest release?

**Comment by Klabost** [comment]
Hey i do not know i have to make a new issue but i have the same problem. 
Sensor values are only updated when the master home assistant is rebooted and then it just stays constant.
There is nothing in the logs about an error or anything. I did put the log output to debug which generated way too much information but I did notice that the heartbeat part of the code was properly working so there is a live connection
